api/lib/infrastructure/repositories/tutorial-repository.js

```diff
@@ -45,9 +45,17 @@ function _toDomain(tutorialData) {
   });
 }
 
-async function _findByRecordIds(ids) {
-  const tutorialDatas = await tutorialDatasource.findByRecordIds(ids);
-  return _.map(tutorialDatas, (tutorialData) => _toDomain(tutorialData));
+function _extractLangFromLocale(locale) {
+  return locale.split('-')[0];
+}
+
+async function _findByRecordIds({ ids, locale }) {
+  let tutorialData = await tutorialDatasource.findByRecordIds(ids);
+  if (locale) {
+    const lang = _extractLangFromLocale(locale);
+    tutorialData = tutorialData.filter((tutorial) => _extractLangFromLocale(tutorial.locale) === lang);
+  }
+  return _.map(tutorialData, (tutorialData) => _toDomain(tutorialData));
 }
```

Comment:
Je pense qu'on peut gagner en lisibilité si on applique la [step down rule](https://dzone.com/articles/the-stepdown-rule) (au moins là) comme suit : 

```suggestion
async function _findByRecordIds({ ids, locale }) {
  let tutorialData = await tutorialDatasource.findByRecordIds(ids);
  if (locale) {
    const lang = _extractLangFromLocale(locale);
    tutorialData = tutorialData.filter((tutorial) => _extractLangFromLocale(tutorial.locale) === lang);
  }
  return _.map(tutorialData, (tutorialData) => _toDomain(tutorialData));
}

function _extractLangFromLocale(locale) {
  return locale.split('-')[0];
}
```